packages/core/admin/ee/server/services/seat-enforcement.js

```diff
@@ -0,0 +1,161 @@
+    where: { id: data.id },
+    data: { value: JSON.stringify(remainingDisabledUsers) },
+  });
+};
+
+const calculateAdminSeatDifference = async (seatsAllowedByLicense) => {
+  const currentAdminSeats = await getService('user').getCurrentActiveUserCount();
+  return currentAdminSeats - seatsAllowedByLicense;
+};
+
+const disableUsersAboveLicenseLimit = async (numberOfUsersToDisable) => {
+  const users = await strapi.db.query('admin::user').findMany({
+    where: { isActive: 'true' },
+    orderBy: { createdAt: 'DESC' },
+    populate: { roles: { select: ['id'] } },
+  });
+
+  const usersToDisable = take(users, numberOfUsersToDisable);
+
+  for await (const user of usersToDisable) {
+    try {
+      await strapi.db.query('admin::user').update({
+        where: { id: user.id },
+        data: {
+          isActive: false,
+        },
+      });
+      user.isActive = false;
+    } catch (error) {
+      return;
+    }
+  }
+
+  const data = await strapi.db.query('strapi::ee-store').findOne({
+    where: { key: 'ee_disabled_users' },
+  });
+
+  if (data) {
+    await strapi.db.query('strapi::ee-store').update({
+      where: { id: data.id },
+      data: { value: JSON.stringify(usersToDisable) },
+    });
+  }
+
+  await strapi.db.query('strapi::ee-store').create({
+    data: {
+      key: 'ee_disabled_users',
+      value: JSON.stringify(usersToDisable),
+    },
+  });
+};
+
+const syncDisabledUserRecords = async () => {
+  const data = await strapi.db.query('strapi::ee-store').findOne({
+    where: { key: 'ee_disabled_users' },
+  });
+
+  if (!data?.value || data.value.length === 0) {
+    return;
+  }
+
+  const disabledUsers = JSON.parse(data.value);
+  disabledUsers.forEach(async (user) => {
+    const data = await strapi.db.query('admin::user').findOne({
+      where: { id: user.id },
+    });
+
+    if (!data) {
+      return;
+    }
+
+    await strapi.db.query('admin::user').update({
+      where: { id: user.id },
+      data: { isActive: false },
+    });
+  });
+};
+
+const revertSeatEnforcementWorkflow = async () => {
+  const data = await strapi.db.query('strapi::ee-store').findOne({
+    where: { key: 'ee_disabled_users' },
+  });
+
+  if (!data?.value || data.value.length === 0) {
+    return;
+  }
+
+  const disabledUsers = JSON.parse(data.value);
+  disabledUsers.forEach(async (user) => {
+    const data = await strapi.db.query('admin::user').findOne({
+      where: { id: user.id },
+    });
+
+    if (!data || data.isActive !== user.isActive) {
+      return;
+    }
+
+    await strapi.db.query('admin::user').update({
+      where: { id: user.id },
+      data: { isActive: true },
+    });
+  });
+};
+
+const seatEnforcementWorkflow = async () => {
+  const permittedAdminSeats = strapi.ee.licenseInfo.seats;
+  if (!permittedAdminSeats) {
+    return;
+  }
+
+  await syncDisabledUserRecords();
+  const adminSeatDifference = await calculateAdminSeatDifference(permittedAdminSeats);
```

Comment:
Sorry to be picky again 🙈 .
Do you think `availableAdminSeatsLeft()` would be a better naming?

Also, maybe getting the permittedAdminSeats could be done inside this function, so you don't need to pass it as an argument.